Signal/src/util/DateUtil.m

```diff
@@ -357,46 +357,50 @@ + (NSString *)formatMessageTimestamp:(uint64_t)timestamp
 
     NSCalendar *calendar = [NSCalendar currentCalendar];
 
+    NSDateComponents *relativeDiffComponents =
+        [calendar components:NSCalendarUnitMinute | NSCalendarUnitHour fromDate:date toDate:nowDate options:0];
+
+    NSInteger minutesDiff = MAX(0, [relativeDiffComponents minute]);
+    if (minutesDiff < 1) {
+        return NSLocalizedString(@"DATE_NOW", @"The present; the current time.");
+    }
```

Comment:
Exit early in the "now" case.

Reply:
Note that this has perf benefits _and_ is a design refinement.   If someone sends or receives a message at Monday 11:59pm and looks at it at Tuesday 12:01 am, we want to see "2 minutes ago", not "Monday".  Myles has confirmed.

Reply:
👍